src/testUtils/jestSetup.js

```diff
@@ -25,6 +25,8 @@ import { utils, HDNode } from 'ethers';
 import StorageMock from './asyncStorageMock';
 import FirebaseMock from './firebaseMock';
 
+process.env['IS_TEST'] = 'TEST'; // eslint-disable-line dot-notation
```

Comment:
when you start jest it sets the process.env.NODE_ENV to 'test', I think you can re-use that. Additionally, you can create a simple function in `utils/environment.js` file to check if the current env is `test`

Reply:
> when you start jest it sets the process.env.NODE_ENV to 'test', I think you can re-use that.

actually no, that's the reason I set it there, because of the following:

When the app starts, it doesn't matter if it's `yarn start` or `yarn test`, `process.env.NODE_ENV` is set to `development` and `test` respectively, however before anything `./utils./shim.js` is being invoked and it fixes `process.env.NODE_ENV=development` so after that in `jest` if you check `process.env.NODE_ENV` it'll be `development`. Even more, you won't be able to modify it, that value is *fixed*. I tested and tried several alternatives and that's the reason here I'm adding a different variable `process.env.IS_TEST` to be able to use TEST environment and conditions.

> Additionally, you can create a simple function in utils/environment.js file to check if the current env is test

Thanks for the suggestion, I'll add that

Reply:
I'd suggest avoiding checking if we're running in test environment as much as we can. I'm trying to understand why this is required, but I'm pretty sure there should be a way of avoiding it.

Reply:
Maybe we can have an environment variable like `USE_REACTOTRON`, for example? (Assuming this is used just for that)

Reply:
> I'd suggest avoiding checking if we're running in test environment as much as we can.

I agree with you when you say "as much as we can" but there're moments when the system should know it's running in a different environment (similar to differentiate between `prod` and `dev` environment) and this is one of those cases.

The idea behind Reactotron is to run anytime you want while you're running your app in `development mode` and check `logs` or the `redux state` while developing, however you don't want to include it (and its dependencies) while you're testing it, for two reasons:

1) it's unnecessary since you're not testing it
2) in this particular case you'll get an error on testing trying to use something outside `nodejs/console` environment as you can see in the following screenshot.

![image](https://user-images.githubusercontent.com/316812/56855185-586abd80-6919-11e9-9e5e-a19cf0eb782c.png)

That crash happens because `App.js` tests how the app loads, and Reactotron is configured and started at that point. You can check that [here](https://github.com/pillarwallet/pillarwallet/pull/1075/files#diff-04298622441e55a9d9b5691873f8490bR34) and [here](https://github.com/pillarwallet/pillarwallet/pull/1075/files#diff-c375b4b4e917a1fddd16650469169fcdR20)

> Maybe we can have an environment variable like USE_REACTOTRON
We could, but once we add that variable into `.env` file we're still under the previous scenario, `jest` runs and `node.process.env` is development (because of the `shim.js` detail [commented above](https://github.com/pillarwallet/pillarwallet/pull/1075#discussion_r279015053)) and we don't have a way to tell jest to not include `Reactotron` dependency.

The thing is, even you're right implying the app shouldn't behave differently in `test`, `dev` or `prod` environment, the importance to know in which environment you are is not excluded, that's important on web and so does on native.